jekyll/_cci2/bundling-config.md

```diff
@@ -0,0 +1,905 @@
+You can optionally invoke a job with one or both of these arguments. Steps under `pre-steps`
+are executed before any of the other steps in the job. The steps under
+`post-steps` are executed after all of the other steps.
+
+For this reason, the parameter names `pre-steps` and `post-steps` are reserved
+and may not be redefined by a job author. For example, the following job
+definition is invalid:
+
+```yaml
+jobs:
+  foo:
+    parameters:
+      pre-steps:    # invalid: pre-steps is a reserved parameter name
+        type: steps
+        default: []
+```
+
+All jobs accept two special arguments of type `steps`: `pre-steps` and
+`post-steps`.
+
+If an orb user invokes a job with one or both of these arguments,
+the job will run the steps in `pre-steps` first, before any other steps run, and
+then it will run the steps in `post-steps` last, after any other steps run.
+
+Pre- and post- steps allow users to be execute steps in a given job's environment
+without modifying the orb. This is useful, for example, when a user imports a job
+and wants to upload assets after it completes, or to run some custom setup steps
+before job execution. Pre- and post- steps allow the user to make these additions
+without modifying the imported job.
+
+A `steps` parameter can be used for a similar purpose, passing steps into a job,
+but it requires that the job be modified with an execution site for the parameter.
+
+#### Example of Using Pre- and Post-Steps
+{:.no_toc}
+
+An orb `foo` might define a job:
+
+```yaml
+# yaml from orb `foo`
+jobs:
+  bar:
+    machine: true
+    steps:
+      - checkout
+      - run:
+          command: echo "building"
+      - run:
+          command: echo "testing"
+```
+
+Then an orb user could use the job as follows:
+```yaml
+# config.yml
+version: 2.1
+orbs:
+  foo: somenamespace/foo@1.2
+workflows:
+  build:
+    jobs:
+      - foo/bar:
+          pre-steps:
+            - run:
+                command: echo "install custom dependency"
+          post-steps:
+            - run:
+                command: echo "upload artifact to s3"
+```
+
+The resulting configuration would look like this:
+
+```yaml
+version: 2.1
+jobs:
+  foo/bar:
+    machine: true
+    steps:
+      - run:
+          command: echo "install custom dependency"
+      - checkout
+      - run:
+          command: echo "building"
+      - run:
+          command: echo "testing"
+      - run:
+          command: echo "upload artifact to s3"
+workflows:
+  build:
+    jobs:
+      - foo/bar
+```
+
+## Conditional Steps
+**Note:** Conditional steps are available in configuration version 2.1 and later.
+
+Conditional steps allow the definition of steps that only run if a condition is
+met. 
+
+<!---
+For example, an orb could define a command that runs a set of steps *if* the
+orb's user invokes it with `myorb/foo: { dostuff: true }`, but not
+`myorb/foo: { dostuff: false }`.
+-->
+
+These conditions are checked before a workflow is actually run. That
```

Comment:
For now we could go further and say that "conditional steps currently only use parameter values as inputs"

Reply:
thanks, added.